doc/arch/adr-001-angular-folder-structure.md

```diff
@@ -0,0 +1,85 @@
+# Angular Folder Structure
+
+## Context
+
+The starting point of this project a series of existing demos implementing a uniformed UI design.
+
+### Considerations
+
+With a strong sense that: 
+- New demo could be added, and each demo should be replaced or removed at ease. 
+- UI components should be reusable in each demo.
+
+We decided to export each demo as an [NgModule](https://angular.io/guide/ngmodule), and the shared UI components a shared module.
+
+In that case, it is easier to extend the current framework with capabilities to switch the demos. And these sub modules can be lazy loaded asynchronously by the router.
+
+## Decisions
+
+We decided to use `shared` as the folder name that houses shared resources (components, directives, etc...).
+
+Considering these changes, the new project structure looks like this:
+
+```
+.
+├── app
+│   ├── dashboard
+│   │   │   ├── dashboard.component.html
+│   │   │   ├── dashboard.component.ts
+│   │   │   └── dashboard.module.ts
+│   ├── demos
+│   │   ├── demo.component.html
+│   │   ├── demos.ts
+│   │   ├── foo
+│   │   │   ├── foo-demo.ts
+│   │   │   ├── foo.component.ts
+│   │   │   ├── foo.module.ts
+│   │   │   └── foo.service.ts
+│   │   └── ...
+│   ├── shared
+│   │   ├── components
```

Comment:
It looks like you nested the components in folders here in the same way that they're nested in the actual HTML of the application. Consider flattening this tree, since there aren't a ton of components, so that each component is just a separate folder under the `shared/components/` folder, such as `shared/components/layer-card/layer-card.component.ts`, etc. This improves discoverability and doesn't tie your physical repo folder structure to the component tree of the application.

See the [Application structure and ngModules](https://angular.io/guide/styleguide#application-structure-and-ngmodules) section of the Angular Style Guide for more reasoning, examples and info.